lib/whatsnew/remote_files.rb

```diff
@@ -44,19 +44,41 @@ def client
       end
 
       def find_from_contents
-        if resources = client.contents(repo)
-          resources.find do |resource|
-            resource.name =~ FILES_TO_SEARCH_REGEXP
-          end
+        resources = client.contents(repo)
+
+        if resources.respond_to?(:message) && resources.message == "Moved Permanently" &&
+           new_location = resources.url
+          search_for_changelog_like_file(new_resources_from(new_location))
+        else
+          search_for_changelog_like_file(resources)
         end
       end
 
+      def new_resources_from(url)
+        http_get(normalize_uri(url))
+      end
+
+      def normalize_uri(url)
+        require "uri"
+        URI(url)
+      end
+
+      def http_get(uri)
+        Dish(JSON.parse(uri.read))
```

Comment:
Would prefer not to pull in another dependency just for `resources.name`.

Also, the extraction of `normalize_uri` doesn't yield more readability than just doing 
`JSON.parse(URI(uri).read)`